src/Collection.php

```diff
@@ -216,13 +214,12 @@ public function __toString()
      * @see Aggregate::__construct() for supported options
      * @param array $pipeline Aggregation pipeline
      * @param array $options  Command options
-     * @return CursorInterface&Iterator
      * @throws UnexpectedValueException if the command response was malformed
      * @throws UnsupportedException if options are not supported by the selected server
      * @throws InvalidArgumentException for parameter/option parsing errors
      * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
      */
-    public function aggregate(array $pipeline, array $options = [])
+    public function aggregate(array $pipeline, array $options = []): CursorInterface&Iterator
```

Comment:
`CursorInterface` already extends `Traversable`, which is better IMO. We should not require the `Iterator` interface for custom cursor classes.
In case people really need the `Iterator` API, they can wrap it in an `IteratorIterator`.
```suggestion
    public function aggregate(array $pipeline, array $options = []): CursorInterface
```

Reply:
A `CursorInterface` type without an intersection with `Iterator` won't allow you to call `$cursor->next();` - that's an issue we faced in ODM. Ideally, we'd fix this in PHPC by having `CursorInterface` extend `Iterator`, but until then we're stuck with this solution.

Reply:
There is always solutions to wrap a Transversable that is not an `Iterator`.
With a Generator (see [UnrewindableIterator](https://github.com/doctrine/mongodb-odm/blob/8e0deb186540a89111181c3be89924a3d791045c/lib/Doctrine/ODM/MongoDB/Iterator/UnrewindableIterator.php#L126-L127]), `IteratorIterator` or `IteratorAggregate::getIterator()`

Reply:
Providing some additional context and throwing in my two cents:

 * We added Iterator to Cursor in [PHPC-1691](https://jira.mongodb.org/browse/PHPC-1691) in order to comply with PHP 8.
 * CursorInterface was introduced in [PHPC-1123](https://jira.mongodb.org/browse/PHPC-1123) to allow other classes to fill in for a Cursor, which cannot be extended. AFAIK, the only existing use case for this within our own driver is CodecCursor.
 * CodecCursor currently implements both CursorInterface and Iterator, so that would not be negatively impacted by this change.

While conditionally wrapping a CursorInterface to get access to Iterator methods is always possible, it may be tedious. Given that CursorInterface is intended to be used for classes that compose a Cursor, it seems a small ask for the wrapping class to also implement Iterator.

I'm not sure what the use case would be for a CursorInterface implementation to rely on IteratorAggregate, as it would then have no way to influence the inner Cursor's iteration. Hypothetically, it could proxy other methods (e.g. `toArray()`), but I've never encountered such an implementation in the wild.

If having CursorInterface extend Iterator doesn't negatively impact PHPLIB, Doctrine ODM, and the Laravel integration, I wouldn't be too concerned about other use cases. Worst case, we're asking users to implement a implement a few extra methods (easily accomplished with a trait if needed).

Reply:
Sounds good. Added PHPC-2447 to track this.

I'll note that we shouldn't implement `IteratorAggregate`, as that would suggest one can obtain multiple iterators from a single cursor. As a cursor can't be rewound and may only ever yield a single iterator, I don't think we should consider implementing `IteratorAggregate`.

Reply:
`CursorInterface` was changed to extend `Iterator`, and this PR now includes changes for PHPLIB-1114, using `CursorInterface` where possible.